templates/master.erb

```diff
@@ -1,29 +1,13 @@
-# ZooKeeper URL (used for leader election amongst masters)
-ZK="<%= @zookeeper %>"
+MESOS_work_dir="<%= @work_dir %>"
+MESOS_zk="<%= @zookeeper %>"
+MESOS_quorum=<%= @quorum %>
+MESOS_ip="<%= @listen_address %>"
+MESOS_port=<%= @master_port %>
+MESOS_cluster="<%= @cluster %>"
+MESOS_log_dir="<%= @log_dir %>"
+MESOS_logbufsecs=1
+MESOS_logging_level="INFO"
 
-# Human readable name for the cluster, displayed in the webui
-CLUSTER="<%= @cluster %>"
-
-# IP address to listen on (in case that you start master
-# on this node, it will bind to this address)
-<% if @listen_address -%>
-IP="<%= @listen_address %>"
```

Comment:
@deric According to the documentation, Mesos takes either MESOS_IP, or MESOS_IP_DISCOVERY_COMMAND. If both are set, Mesos will throw an error. This is an attempt to be able to pass any of them here (we pass in MESOS_IP_DISCOVERY_COMMAND as extra parameters).